File: scripts/fetch_block_header.js
async function fetchBlockHeader(blockHeight) {
    try {
        // First get the block hash from height
        const heightResponse = await fetch(`https://mempool.space/api/block-height/${blockHeight}`);
        const blockHash = await heightResponse.text();
        
        // Then get the block header using the hash
        const headerResponse = await fetch(`https://mempool.space/api/block/${blockHash.trim()}/header`);
        const headerHex = await headerResponse.text();
        
        console.log(`Block Height: ${blockHeight}`);
        console.log(`Block Hash: ${blockHash.trim()}`);
        console.log(`Block Header (hex): ${headerHex.trim()}`);
        console.log(`Header Length: ${headerHex.trim().length / 2} bytes`);
        
        return headerHex.trim();
    } catch (error) {
        console.error('Error fetching block header:', error);
    }
}

// Example usage
const blockHeight = 841968; // You can change this to any height
fetchBlockHeader(blockHeight);

File: scripts/fetch_block_header.py
import requests

def fetch_block_header(block_height):
    # First get the block hash from height
    height_url = f"https://mempool.space/api/block-height/{block_height}"
    block_hash = requests.get(height_url).text.strip()
    block_hash_le = ''.join(reversed(block_hash))

    # Then get the block header using the hash
    header_url = f"https://mempool.space/api/block/{block_hash}/header"
    header_hex = requests.get(header_url).text.strip()
    
    print(f"Block Height: {block_height}")
    print(f"Block Hash: {block_hash_le}")
    print(f"Block Header (hex): {header_hex}")
    print(f"Header Length: {len(block_hash_le)} bytes")
    
    return header_hex

# Example usage
if __name__ == "__main__":
    
    fetch_block_header(897110)
    fetch_block_header(895104)



